### PR TITLE
Alexlovelltroy/build improvements

### PR DESCRIPTION
**Pull request by alexlovelltroy**
Adding build automation from the work I did for smd that leads to better version reporting, multiarch containers, and native packages

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,6 @@
 kubernetes/.packaged/
 .idea/
 *.swp
-.version
-
-# Binaries
-boot-script-service
```
Comment (davidallendj) [inline .gitignore:7]: Why remove the binaries from the `.gitignore`?

File: cmd/boot-script-service/version.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	fmt.Printf("Build Time: %s\n", BuildTime)
+	fmt.Printf("Git Branch: %s\n", GitBranch)
+	fmt.Printf("Git Tag: %s\n", GitTag)
+	fmt.Printf("Git State: %s\n", GitState)
+	fmt.Printf("Build Host: %s\n", BuildHost)
+	fmt.Printf("Go Version: %s\n", GoVersion)
+	fmt.Printf("Build User: %s\n", BuildUser)
+}
+
+func VersionInfo() string {
```
Comment (davidallendj) [inline cmd/boot-script-service/version.go:55]: Is this function necessary? I don't see where it's being used.

**Review by davidallendj — COMMENTED** [review]
I made two inquiries about the changes, but neither are necessarily blockers.

File: cmd/boot-script-service/version.go (hunk 2 above)
Comment (davidallendj) [reply cmd/boot-script-service/version.go:55]: Resolving based on https://github.com/OpenCHAMI/magellan/pull/66#issuecomment-2447810343.

File: .gitignore (hunk 1 above)
Comment (alexlovelltroy) [reply .gitignore:7]: The binaries don't live there anymore and it was catching other files that should be part of the repo

**Comment by synackd** [comment]
We should probably also update the README with build instructions using goreleaser (e.g. setting the env vars).

**Comment by synackd** [comment]
Commenting here for posterity: `.Arm64` was introduced in Goreleaser v2.4 so at least that Goreleaser version is required (see: https://goreleaser.com/customization/templates/#single-artifact-extra-fields).

Otherwise one will get the following error:
```
  ⨯ release failed after 2m34s               error=template: failed to apply "dist/bss_{{ .Os }}_{{ if eq .Arch \"amd64\" }}{{ .Arch }}_{{ .Amd64 }}{{ else if eq .Arch \"arm64\" }}{{ .Arch }}_{{ .Arm64 }}{{ else }}{{ .Arch }}{{ end }}/boot-script-service": map has no entry for key "Arm64"
```

**Review by synackd — COMMENTED** [review]
The build works using Goreleaser >=2.4 and I was able to test the amd64 container. I haven't been able to test arm64 yet.

File: .github/workflows/build_release.yml [hunk 3]
```diff
@@ -54,16 +62,27 @@ jobs:
           echo "fs.writeFileSync('digest.txt', firstNonNullDigest);" >> process.js
           node process.js
           echo "digest=$(cat digest.txt)" >> $GITHUB_OUTPUT
-      - name: Attest boot-script-service binary
-        uses: github-early-access/generate-build-provenance@main
+
+      - name: Attest boot-script-service binary amd64
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-path: dist/bss_linux_amd64_v4/boot-script-service
```
Comment (synackd) [inline .github/workflows/build_release.yml:69]: v4 -> v3 here and line 77

**Review by synackd — CHANGES_REQUESTED** [review]
Tested goreleaser on the same amd64 machines and worked on all of them.

I obviously couldn't test the GitHub actions stuff, but found a small thing that should be fixed before merging.

Also, can we add the environment variable setting instructions for Goreleaser to the README (e.g. [this](https://github.com/OpenCHAMI/smd/blob/a66ffe1af5f1ce8b3a88b65c6927bbe2b3f306e7/README.md?plain=1#L15-47))? I think the instructions are helpful for building locally, but also it's useful to, when building locally, do something like
```sh
eval $(grep ^export README.md)
```
to automatically source those variables.

**Review by synackd — APPROVED** [review]
Everything is resolved, LGTM.

**Pull request merged**